### ISSUE TITLE
[Package Request] - haveged

### ISSUE THREAD

**Issue by adrianmace** [amazonlinux/amazon-linux-2023#278]
**What package is missing from Amazon Linux 2023? Please describe and include package name.**
haveged

**Is this an update to existing package or new package request?**
New package request

**Is this package available in Amazon Linux 2?**
No.

**Number of users impacted**
?

**Comment by stewartsmith** [comment]
I'm gathering this is https://src.fedoraproject.org/rpms/haveged which is:

> A Linux entropy source using the HAVEGE algorithm
>
>Haveged is a user space entropy daemon which is not dependent upon the
>standard mechanisms for harvesting randomness for the system entropy
>pool. This is important in systems with high entropy needs or limited
>user interaction (e.g. headless servers).
>
>Haveged uses HAVEGE (HArdware Volatile Entropy Gathering and Expansion)
>to maintain a 1M pool of random bytes used to fill /dev/random
>whenever the supply of random bits in /dev/random falls below the low
>water mark of the device. The principle inputs to haveged are the
>sizes of the processor instruction and data caches used to setup the
>HAVEGE collector. The haveged default is a 4kb data cache and a 16kb
>instruction cache. On machines with a cpuid instruction, haveged will
> attempt to select appropriate values from internal tables. ([upstream](https://github.com/jirka-h/haveged))

What entropy issues are you running into with Amazon Linux 2023?

**Comment by adrianmace** [comment]
Yep, that’s correct.

We are not seeing any issues because we haven’t run production workloads on AL2023 yet.

For me, this is a “nice to have”.

**Comment by daniejstriata** [comment]
I did some research (and waiting on CIS benchmarks for AL2023 to confirm) finding that since kernel 5.6, haveged does not make any noticeable improvement to a hosts entropy. Seems that where it could still help is to provide entropy early when the host is booting. Having an additional source of entropy is not a bad thing though. 
https://github.com/jirka-h/haveged/issues/57

**Comment by ozbenh** [comment]
The kernel has the ability to get jitter entropy, and entropy from the firmware seeded from a hardware entropy source. Some instance types (most x86 and Graviton 3) additionally have a user accessible hwrng which the kernel also exploits to obtain entropy.

**Comment by adrianmace** [comment]
Yep. I am actually in favour of closing this request now to honour the most up to date best practices.

Will leave for the maintainers to close.
